mlflow/store/artifact/artifact_repo.py

```diff
@@ -189,17 +189,16 @@ def _download_file(src_artifact_path, dst_local_dir_path):
 
         # Wait for downloads to complete and collect failures
         failed_downloads = {}
-        for f in (
-            pbar := ArtifactProgressBar.files(
-                as_completed(futures), desc="Downloading artifacts", total=len(futures)
-            )
-        ):
-            try:
-                f.result()
-                pbar.update = True
-            except Exception as e:
-                path = futures[f]
-                failed_downloads[path] = repr(e)
+        with ArtifactProgressBar.files(
+            as_completed(futures), desc="Downloading artifacts", total=len(futures)
+        ) as pbar:
+            for f in pbar:
+                try:
+                    f.result()
+                    pbar.update = True
+                except Exception as e:
+                    path = futures[f]
+                    failed_downloads[path] = repr(e)
```

Comment:
```suggestion
        with ArtifactProgressBar.files(
            desc="Downloading artifacts", total=len(futures)
        ) as pbar:
            for f in as_completed(futures):
                try:
                    f.result()
                    pbar.upate()
                except Exception as e:
                    path = futures[f]
                    failed_downloads[path] = repr(e)
```

I think we no longer need to pass `as_completed(futures)` to `ArtifactProgressBar`.

Reply:
Why? Aren't we iterate on the ArtifactProgressBar in order to call .update and .refresh on the original tqdm bar?

Reply:
I'm not a big fan of switching the behavior by assigning a value to `pbar.update`.